integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsHandlingNullJsonObject.xml

```diff
@@ -0,0 +1,40 @@
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+  ~
+  ~ WSO2 Inc. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied. See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<api xmlns="http://ws.apache.org/ns/synapse" context="/nashornJsHandlingNullJsonObject" name="nashornJsHandlingNullJsonObject">
+    <resource methods="POST">
+        <inSequence>
+            <script language="nashornJs"><![CDATA[
+
+		mc.setJsonText(mc, null);
+
+		var jsonText = mc.getJsonText();
+		var parsedJsonText = JSON.parse(jsonText);
+		mc.setJsonObject(mc, parsedJsonText);
+
+		var retrivedJsonObject = mc.jsonObject(mc);
+		mc.setProperty("json", retrivedJsonObject);
+
+		var jsonproperty = mc.getProperty("json");
+		mc.setPayloadJSON(jsonproperty);
+
+	   ]]></script>
+            <respond/>
+        </inSequence>
+    </resource>
+</api>
```

Comment:
Better to have a new line

Reply:
Hi Sumedha,
Fixed with following PR #1511